### PR TITLE
Add Security Gallery entry for new Dependency Review Action

### PR DESCRIPTION
**Pull request by elireisman**
As stated in the title. I cut this PR locally before having access to the `starter-workflows` repo so I didn't see this PR template and instructions until now. I will review and make updates as needed  before taking this out of draft, but feel free to call out anything I missed!

The goal is to prep a PR for merge that adds the new `actions/dependency-review-action` repo to the Security Gallery tab. We plan to create that repo by EOW and move our staff-shipped code from [here](https://github.com/dsp-testing/dependency-review-action) to there after a sanitation pass to prep the Git history (etc) for GA and open sourcing 👍 

## Pre-requisites

- [ ] Prior to submitting a new workflow, please apply to join the GitHub Technology Partner Program: [partner.github.com/apply](https://partner.github.com/apply?partnershipType=Technology+Partner).

## Tasks

**For _all_ workflows, the workflow:**

- [x] Should be contained in a `.yml` file with the language or platform as its filename, in lower, [_kebab-cased_](https://en.wikipedia.org/wiki/Kebab_case) format (for example, [`docker-image.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-image.yml)).  Special characters should be removed or replaced with words as appropriate (for example, "dotnet" instead of ".NET").
- [x] Should use sentence case for the names of workflows and steps (for example, "Run tests").
- [x] Should be named _only_ by the name of the language or platform (for example, "Go", not "Go CI" or "Go Build").
- [x] Should include comments in the workflow for any parts that are not obvious or could use clarification.
- [x] Should specify least priviledge [permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token) for `GITHUB_TOKEN` so that the workflow runs successfully. 

**For _CI_ workflows, the workflow:**

- [ ] Should be preserved under [the `ci` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [ ] Should include a matching `ci/properties/*.properties.json` file (for example, [`ci/properties/docker-publish.properties.json`](https://github.com/actions/starter-workflows/blob/main/ci/properties/docker-publish.properties.json)).
- [ ] Should run on `push` to `branches: [ $default-branch ]` and `pull_request` to `branches: [ $default-branch ]`.
- [ ] Packaging workflows should run on `release` with `types: [ created ]`.
- [ ] Publishing workflows should have a filename that is the name of the language or platform, in lower case, followed by "-publish" (for example, [`docker-publish.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-publish.yml)).

**For _Code Scanning_ workflows, the workflow:**

- [x] Should be preserved under [the `code-scanning` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [x] Should include a matching `code-scanning/properties/*.properties.json` file (for example, [`code-scanning/properties/codeql.properties.json`](https://github.com/actions/starter-workflows/blob/main/code-scanning/properties/codeql.properties.json)), with properties set as follows:
  - [x] `name`: Name of the Code Scanning integration.
  - [ ] `organization`: Name of the organization producing the Code Scanning integration.
  - [x] `description`: Short description of the Code Scanning integration.
  - [x] `categories`: Array of languages supported by the Code Scanning integration.
  - [x] `iconName`: Name of the SVG logo representing the Code Scanning integration. This SVG logo must be present in [the `icons` directory](https://github.com/actions/starter-workflows/tree/main/icons).
- [x] Should run on `push` to `branches: [ $default-branch, $protected-branches ]` and `pull_request` to `branches: [ $default-branch ]`. We also recommend a `schedule` trigger of `cron: $cron-weekly` (for example, [`codeql.yml`](https://github.com/actions/starter-workflows/blob/c59b62dee0eae1f9f368b7011cf05c2fc42cf084/code-scanning/codeql.yml#L14-L21)).

**Some general notes:**

- [x] This workflow must _only_ use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**
- [ ] This workflow must _only_ use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  We require that these actions be referenced using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
    ```
    # This workflow uses actions that are not certified by GitHub.
    # They are provided by a third-party and are governed by
    # separate terms of service, privacy policy, and support
    # documentation.
    ```
- [x] Automation and CI workflows should not send data to any 3rd party service except for the purposes of installing dependencies.
- [x] Automation and CI workflows cannot be dependent on a paid service or product.

### CODE REVIEW

File: dependency-graph/dependency-review.yml
Comment (elireisman) [inline dependency-graph/dependency-review.yml:11]: TODO: before GA next week, when we plan to merge this PR, it looks from the PR template docs that we'll need to replace `main` tag with a fixed commit SHA?
1

Can you confirm this is the case? If so once the `actions/dependency-review-action` repo is released we can update to a fixed SHA/tagged release as needed 👍

File: dependency-graph/dependency-review.yml
Comment (elireisman) [inline dependency-graph/dependency-review.yml:2]: Our Action is not quite the same as those described in the PR template under `code-scanning` (the closest match IMO) our Action needs to run against PRs on push to topic branches, not `$default_branch`.

See the staff-ship repo linked in the PR description for details, and here's an example of a staff-ship installed repo where you can see the Action at work: [link](https://github.com/github/dependency-graph-api/actions/workflows/dependency-review.yml)

File: dependency-graph/properties/dependency-review.properties.json
Comment (elireisman) [inline dependency-graph/properties/dependency-review.properties.json:17]: I added tags that are not specific to language ecosystems, since our Action again is not a 1:1 match with `code-scanning` as described in PR template. Do I need to update this?

**Review by elireisman — COMMENTED** [review]
Generally, I need to ask reviewers if our new DR Action is best registered under `code-scanning/` as described in the PR template, or can exist on it's own since the behavior (workflow triggers etc.), purpose, and team ownership is also different.

I'm happy to do whatever makes this smooth for reviewers and maintainers, and gets our new DR Action into the Security Gallery in time for GA, just not sure where this new Action fits yet 👍 guidance welcome/encouraged!

File: dependency-graph/properties/dependency-review.properties.json
Comment (bishal-pdMSFT) [inline dependency-graph/properties/dependency-review.properties.json:3]: No need to have creator if you want to show up as `By GitHub`.

File: dependency-graph/dependency-review.yml
Comment (bishal-pdMSFT) [reply dependency-graph/dependency-review.yml:2]: Do you want this workflow for PR against any branch? That would be lots of run without much value. That's why we usually use `$default_branch` as that's the important branch.

File: dependency-graph/properties/dependency-review.properties.json
Comment (bishal-pdMSFT) [inline dependency-graph/properties/dependency-review.properties.json:15]: Why `Actions`?

File: dependency-graph/dependency-review.yml
Comment (bishal-pdMSFT) [reply dependency-graph/dependency-review.yml:11]: Commit SHA is not needed for first party actions. You can use a version tag though instead of `main`

File: dependency-graph/dependency-review.yml
Comment (bishal-pdMSFT) [inline dependency-graph/dependency-review.yml:4]: Please add the [permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token) which this job would need.

File: dependency-graph/properties/dependency-review.properties.json
Comment (bishal-pdMSFT) [inline dependency-graph/properties/dependency-review.properties.json:7]: I am not too sure if this starter workflow would show up in `Security` section. Not sure but I think that is hard coupled today with `Code scanning` category.
@anuragc617 @Phantsure can you please confirm. If so, can we quickly fix this in the UI to allow this to show up.

File: dependency-graph/properties/dependency-review.properties.json
Comment (bishal-pdMSFT) [inline dependency-graph/properties/dependency-review.properties.json:16]: I assume you have covered all the languages which are supported by DR action?

File: dependency-graph/dependency-review.yml
Comment (bishal-pdMSFT) [inline dependency-graph/dependency-review.yml:1]: Typically we suggest few things as comments at top of the workflow file:
- What is the value provided by this workflow
- Any pre-reqs whicih must be met for the workflow to have successful runs - think some assumptions, secrets, external entities which may be needed by Action
- Any doc/support link where user can reach out to read more

Comment (bishal-pdMSFT) [reply dependency-graph/dependency-review.yml:1]: The intent essentially is that user feel comfortable with the workflow. And understands enough to modify it for their use case if needed.

Comment (bishal-pdMSFT) [inline dependency-graph/dependency-review.yml:1]: I see a new directory has been created for `dependency-graph`. While I agree that `code-scnaning` directory does not make sense but we also would not want the directories to proliferate.
@anuragc617 @Phantsure would it be okay if we rename `code-scanning` dir to `security` and keep both together.

File: dependency-graph/dependency-review.yml
Comment (elireisman) [reply dependency-graph/dependency-review.yml:2]: Yes we want this to run against any PR that's correct. The goal is for the Action to fail when a PR is introducing new known-vulnerable package versions in manifest file changes on the repo 👍

File: dependency-graph/dependency-review.yml
Comment (elireisman) [reply dependency-graph/dependency-review.yml:11]: cc @febuiles @hmaurer @lseppala as per open-source guidelines, are we obligated to semver-tag changes to the DR Action so end users can track fixed versions, or can we rely on `main` here to propagate bugfixes/DR Action changes to all installs?

🤔 The former feels like the right play, but the latter will help us avoid a long-tail support burden of folks who aren't upgrading regularly

File: dependency-graph/dependency-review.yml
Comment (elireisman) [reply dependency-graph/dependency-review.yml:4]: My understanding as per [this](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#how-the-permissions-are-calculated-for-a-workflow-job) and discussions with the team is, our `GITHUB_TOKEN` for post-GA installs will only require `repo: read` permissions and might be fine with the `GITHUB_TOKEN` defaults?

cc @febuiles @hmaurer to confirm ?

File: dependency-graph/properties/dependency-review.properties.json
Comment (elireisman) [reply dependency-graph/properties/dependency-review.properties.json:15]: We pick up Actions dependencies declared on repos that include a `.github/workflows` dir and detect vulnerable versions, same as with other language ecosystem packages

File: dependency-graph/properties/dependency-review.properties.json
Comment (elireisman) [reply dependency-graph/properties/dependency-review.properties.json:3]: Excellent, can update 👍

File: dependency-graph/properties/dependency-review.properties.json
Comment (elireisman) [reply dependency-graph/properties/dependency-review.properties.json:15]: That said, I'll take another pass to see if there are more appropriate tags we should include anyway thanks for the callout 🙇

File: dependency-graph/dependency-review.yml
Comment (bishal-pdMSFT) [reply dependency-graph/dependency-review.yml:4]: It would be good practice to nonetheless put "repo: read" permission as this explicitly declares that this workflow only needs such permission. Even if org/repo admin change the default GITHUB_TOKEN to be more permissive or restrictive, the workflow will have just the right level of security.

File: dependency-graph/dependency-review.yml
Comment (bishal-pdMSFT) [reply dependency-graph/dependency-review.yml:1]: @elireisman for now keep the DR starter workflow inside `code-scanning` directory. This will allow it to get shipped. And then we can follow up renaming directory to `security` and updating the UI to work with this directory.

Comment (elireisman) [reply dependency-graph/dependency-review.yml:1]: cc @courtneycl @saritai for docs links 🙇

File: dependency-graph/properties/dependency-review.properties.json
Comment (elireisman) [reply dependency-graph/properties/dependency-review.properties.json]: Got a reply below, moved the files to `code-scanning/` top-level dir tree. Let me know if there's more related changes to enable this (temporary) home for security gallery entries 👍

File: dependency-graph/dependency-review.yml
Comment (elireisman) [reply dependency-graph/dependency-review.yml]: thanks - just moved my action spec and props files into the `code-scanning/` and `code-scanning/properties` dirs. Let me know if there's more to do there related to this move 👍

Comment (courtneycl) [reply dependency-graph/dependency-review.yml]: Doc will live here: https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-dependency-review#dependency-review-enforcement

File: dependency-graph/dependency-review.yml
Comment (lseppala) [reply dependency-graph/dependency-review.yml]: @elireisman as discussed, since the ref used by an Action isn't required to be immutable yet (that is, you can use a branch rather than a tag), I think we should accommodate semver _and_ also have a ref that can be updated with bugfixes. To keep it simple, we can call it `v1`

@febuiles, this will require a small amount of work in the Action repo. We'll need a _branch_ named `v1` to match `main`, which is what this Action will track.

File: dependency-graph/dependency-review.yml
Comment (elireisman) [reply dependency-graph/dependency-review.yml]: We only want to run against `pull_request` so never default branch, only topic branch of PRs on the repo where the Action is installed 👍

File: dependency-graph/dependency-review.yml
Comment (elireisman) [reply dependency-graph/dependency-review.yml]: I'll update the comment now

File: dependency-graph/dependency-review.yml
Comment (elireisman) [reply dependency-graph/dependency-review.yml]: OK will do! thank you!

Comment (elireisman) [reply dependency-graph/dependency-review.yml]: done 👍 thank you!

**Comment by elireisman** [comment]
👋 sanity check - I'm assuming we don't have to do this? `Prior to submitting a new workflow, please apply to join the GitHub Technology Partner Program` from the PR template pre-reqs?

File: dependency-graph/dependency-review.yml
Comment (hmaurer) [reply dependency-graph/dependency-review.yml]: @lseppala @febuiles c.f. https://docs.github.com/en/actions/creating-actions/about-custom-actions#using-tags-for-release-management

File: code-scanning/properties/dependency-review.properties.json
Comment (bishal-pdMSFT) [inline code-scanning/properties/dependency-review.properties.json:6]: You can replaced this with "Dependency Review" which is more appropriate. I have a PR which will allow that to be processed. Will most probably merge that tomorrow my day time.

**Review by bishal-pdMSFT — APPROVED** [review]
LGTM. Will merge once we are ready to hsip.

File: code-scanning/properties/dependency-review.properties.json
Comment (anuragc617) [inline code-scanning/properties/dependency-review.properties.json:7]: I don't think you should have Automation as categories for this templates. Automation templates are added to the automation folder. Also, move "Dependency Review" to the start of categories list.

File: code-scanning/properties/dependency-review.properties.json
Comment (elireisman) [reply code-scanning/properties/dependency-review.properties.json:7]: Sorry - did you mean the `name: "Dependency Review"` or the `categories` tag?

In the tag case, I see other two-word tags with both caps (and Dependency Review is the feature, we captialize it in other docs)

In the "name" case, the other YAML file also names this `Dependency Review` - will all those need updates as well if we change this `name`?

basically - what exactly should I update and in which file to match up with your PR change? Also - if it's easier for you to make a push to this PR directly to make the update feel free to do so, as I may not here back before tomorrow (our target ship date!) due to async/ time zones 🙇 

thanks again for the callout!

File: code-scanning/properties/dependency-review.properties.json
Comment (elireisman) [reply code-scanning/properties/dependency-review.properties.json]: moved "Dependency Review" to top of `categories` array, and removed "Automation" 👍 thanks!

**Pull request merged**
